data_utils/process_waymo_files.py

```diff
@@ -245,25 +294,60 @@ def waymo_to_scenario(
     i = 0
     for dynamic_map_state in protobuf.dynamic_map_states:
         traffic_light_dict = _init_tl_object(dynamic_map_state)
+        # there is a traffic light but we don't want traffic light scenes so just return
+        if len(traffic_light_dict) > 0:
+            return
         for id, value in traffic_light_dict.items():
             for key in all_keys:
                 tl_dict[id][key].append(value[key])
             tl_dict[id]["time_index"].append(i)
         i += 1
 
-    # Construct the object states
-    objects = []
-    for track in protobuf.tracks:
-        obj = _init_object(track)
-        if obj is not None:
-            objects.append(obj)
-
-    # Construct the map states
+    # Construct the map states 
     roads = []
+    edge_segments = []
     for map_feature in protobuf.map_features:
         road = _init_road(map_feature)
         if road is not None:
             roads.append(road)
+            if road["type"] == "road_edge":
+                edge_vertices = [[r["x"], r["y"], r["z"]] for r in road["geometry"]]
+                edge_segments += [[edge_vertices[i], edge_vertices[i+1]] for i in range(len(edge_vertices) - 1)]
+
+    # Construct road edges for collision checking
+    edge_segments = _filter_small_segments(edge_segments)
+    edge_mesh = _generate_mesh(edge_segments)
+    collision_manager = trimesh.collision.CollisionManager()
+    collision_manager.add_object('road_edges', edge_mesh)
+
+    # Construct the object states
+    objects = []
+    for track in protobuf.tracks:
+        obj = _init_object(track)
+        if obj is not None:
+            if obj['type'] not in ['vehicle', 'cyclist']:
+                obj["mark_as_expert"] = False
+                objects.append(obj)
+                continue
+            elif False in obj["valid"]:
+                # Create trajectory segments of only valid positions
+                trajectory_segments = []
+                for i in range(len(obj["position"]) - 1):
+                    if obj["valid"][i] and obj["valid"][i + 1]:
+                        trajectory_segments.append([[obj["position"][i]["x"], obj["position"][i]["y"], obj["position"][i]["z"]],
+                                                    [obj["position"][i+1]["x"], obj["position"][i+1]["y"], obj["position"][i+1]["z"]]])
+            else:
+                obj_vertices = [[pos["x"], pos["y"], pos["z"]] for pos in obj["position"]]
+                trajectory_segments = [[obj_vertices[i], obj_vertices[i+1]] for i in range(len(obj_vertices) - 1)]
+
+            trajectory_segments = _filter_small_segments(trajectory_segments)
```

Comment:
oh huh what's going on here?

Reply:
3D geometry operations for cubes that are smaller than 1e-6 in length lead to a bunch of error values, so this just screens the road edges and object trajectories for those cubes and filters them out